### PR TITLE
Dagger android

### PR DESCRIPTION
**Pull request by alvarez-fabian**
# Summary

Update `wolmo-networking` to make it compatible with dagger.

* Add dagger dependencies as provided.
* Create new `NetworkComponent` to build and configure `RetrofitServices`.
* Add `GsonModule` to provide a `Gson` instance and dependencies to build it.
* Add `NetworkingModule` to provide a `Retrofit` instance.
* Add `OkHttpClientModule` to provide `OkHttpClient`, and add interceptors to it.
* Remove `NetworkingApplication` that's not needed anymore.
* Update `RetrofitServices` to move all the creation and configuration to the specific modules.

# Description

Applications that need to use `wolmo-network` doesn't need anymore to extend `NetworkingApplication`.
This library provides the modules to build and configure the dependencies of retrofit, Gson and OkHttp, it also provides a default `NetworkingComponent` to simplify the creation and reduce configuration on common scenarios. For it to work you only need to provide it the `baseUrl`, an optional list of interceptors to add to `okHttp` and an optional list of `gson` type adapters.

We can find the following modules:

* GsonModule:
* OkHttpClientModule
* NetworkingModule

`NetworkingComponent` is a template component for simple uses, if the clients want to further customize it they can make a new Component and include the modules provided with this library or make their own.
If you think we can separate this modules more please let me know so we can make this more modular.

All the modules and the component in this library uses `@NetworkingScope` to define a common scope.

# TODO
* We should start adding unit tests.
* In the near future we should add some integration testing too. I don't think we need specific android instrumented tests here, since this library does not have any specific dependency on android life cycle.

### CODE REVIEW

File: networking/build.gradle [hunk 1]
```diff
@@ -10,26 +10,34 @@ android {
...
+    ext.retrofit_version = '2.2.0'
+    ext.okhttp_version = '3.7.0'
+}
+
 dependencies {
     compile fileTree(dir: 'libs', include: ['*.jar'])
 
     // Wolmo
-    compile 'com.github.Wolox:wolmo-core-android:v1.0.0'
+    compile 'com.github.Wolox:wolmo-core-android:dagger-android-SNAPSHOT'
```
Comment (emalamela) [inline networking/build.gradle]: We should change the `dagger-android-SNAPSHOT` to the corresponding version when merging the PR. Maybe put a `TODO` :thinking:?

File: networking/src/main/java/ar/com/wolox/wolmo/networking/utils/LoggingUtils.java [hunk 2]
```diff
@@ -19,41 +19,40 @@
...
+import okhttp3.OkHttpClient;
+import okhttp3.logging.HttpLoggingInterceptor;
 
 /**
- * This class extends {@link ar.com.wolox.wolmo.core.WolmoApplication} to easily initialize and get and instance of
- * {@link RetrofitServices} to perform API calls.
+ * Utils to create {@link HttpLoggingInterceptor} for {@link OkHttpClient}.
  */
-public abstract class NetworkingApplication extends WolmoApplication {
+public class LoggingUtils {
```
Comment (emalamela) [inline networking/src/main/java/ar/com/wolox/wolmo/networking/utils/LoggingUtils.java]: Should we build a class just to return an `HttpLoggingInterceptor`?

If it provided several different `Interceptor`s or something like that, i would be more in favor. But i believe this code can be inlined.

**Review by SpicyCactuar — CHANGES_REQUESTED** [review]
Just a couple of comments, but it looks cleeeeeeeean!

File: networking/build.gradle (hunk 1 above)
Comment (alvarez-fabian) [reply networking/build.gradle]: Nice catch, I'll add it :smile:

File: networking/src/main/java/ar/com/wolox/wolmo/networking/utils/LoggingUtils.java (hunk 2 above)
Comment (alvarez-fabian) [reply networking/src/main/java/ar/com/wolox/wolmo/networking/utils/LoggingUtils.java]: I think I'll remove this class and let the client to create and configure a logger interceptor if they want. Do you think we should keep the library `com.squareup.okhttp3:logging-interceptor`?

**Comment by alvarez-fabian** [comment]
@emalamela Done :smile:

**Review by SpicyCactuar — APPROVED** [review]
Good job!

**Comment by alvarez-fabian** [comment]
[Migrar wolmo-networking a dagger](https://trello.com/c/1oNsxOB2/27-migrar-wolmo-networking-a-dagger)

**Pull request merged**